### PR TITLE
Fixes #61: Added an option to create user on the dashboard's "User" section

### LINKED ISSUE #61: "Create user" button on the dashboard
Add a create user button on the dashboard. (only supports email/password or magic link login)

### PR DESCRIPTION
**Pull request by kabir276**
Fixes Issue #61 

Description
I've added an option to create users on the dashboard's "Users" section. I've created a pull request for your review. Please let me know if this is what you envisioned. If not, I'm happy to make further adjustments.

Changes Made 
Updated users>page.client.tsx , lib>stack-app.ts , Interface>clientInterface.ts to include the button to create users.
Additional Notes
No new dependencies were added.

### CODE REVIEW

File: packages/stack-server/src/app/(main)/(protected)/projects/[projectId]/users/page-client.tsx [hunk 1]
```diff
@@ -1,24 +1,111 @@
...
+  setcurrentUserPassword: React.Dispatch<React.SetStateAction<string>>;
+
+};
 
 export default function PageClient() {
   const stackAdminApp = useAdminApp();
   const allUsers = stackAdminApp.useServerUsers();
+  const [addUserOpen, setAddUserOpen] = React.useState(false);
+  const [showNotifyPasswordDialog, setShowNotifyPasswordDialog] = React.useState(false);
+  const [currentUserPassword, setcurrentUserPassword] = React.useState("");
```
Comment (fomalhautb) [inline packages/stack-server/src/app/(main)/(protected)/projects/[projectId]/users/page-client.tsx]: `setcurrent...` -> `setCurrent...`

File: packages/stack-server/src/app/(main)/(protected)/projects/[projectId]/users/page-client.tsx [hunk 2]
```diff
@@ -1,24 +1,111 @@
...
+        setcurrentUserPassword={setcurrentUserPassword}
+      />
+      <ActionDialog
+        title="Password Change Required"
+        okButton={{
+          label: "Change Password Soon",
+          onClick: handlePasswordNotificationClose,
+        }}
+        open={showNotifyPasswordDialog}
+        onClose={handlePasswordNotificationClose}
```
Comment (fomalhautb) [inline packages/stack-server/src/app/(main)/(protected)/projects/[projectId]/users/page-client.tsx]: eslint not passing: you should use `runAsynchronouslyWithAlert`, check out the other occurrence. Also please add the eslint plugin to vscode so the errors will be highlighted

File: packages/stack/src/lib/stack-app.ts [hunk 3]
```diff
@@ -1019,6 +1020,23 @@ class _StackClientAppImpl<HasTokenStore extends boolean, ProjectId extends strin
...
+  }): Promise<string | KnownErrors["UserEmailAlreadyExists"] | KnownErrors['PasswordRequirementsNotMet'] | undefined> {
+    const emailVerificationRedirectUrl = constructRedirectUrl(this.urls.emailVerification);
+    const password =generateSecureRandomString()
+    const errorCode = await this._interface.signUpWithCredential(
+      options.email,
+      password,
+      emailVerificationRedirectUrl,
+      null,
+    );
+    if (!errorCode) {
```
Comment (fomalhautb) [inline packages/stack/src/lib/stack-app.ts:1034]: This is a promise, you need to add await

**Review by fomalhautb — COMMENTED** [review]
I get 404 API key not found error when I try to create a user on the dashboard. Did it work for you?

**Comment by kabir276** [comment]
Yes, I was getting the same issue but I explicitly commented that “if statement” causing the issue and asked @N2D4  about it. He asked me to make a PR and he’ll take a look at it.

**Comment by fomalhautb** [comment]
The PR seems to be inactive

**Pull request closed without merging**